servant-client/src/Servant/Client/Internal/HttpClient.hs

```diff
@@ -185,11 +185,11 @@ performRequest req = do
         now <- getCurrentTime
         bss <- Client.brConsume $ Client.responseBody fRes
         let fRes' = fRes { Client.responseBody = BSL.fromChunks bss }
-        mapM_ (updateCookieJar now) $ Client.hrRedirects responses <> [(fReq, fRes')]
+        mapM_ (atomically . updateCookieJar now) $ Client.hrRedirects responses <> [(fReq, fRes')]
```

Comment:
`atomically` on the outside. We want single STM transaction, don't we?

Reply:
Yes, you're correct. We only want to do one STM transaction per chain of request/responses. Fixed.